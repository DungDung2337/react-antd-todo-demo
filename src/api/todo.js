import axiosClient from './axiosClient';
const url = '/api/todo';

const todoApi = {
  getAll(params) {
    return axiosClient.get(url, { params });
  },

  addTodoItem(data) {
    return axiosClient.post(url, data);
  },

  updateTodoItem(id, data) {
    const urlWithId = `${url}/${id}`;
    return axiosClient.put(urlWithId, data);
  },

  deleteTodoItem(id) {
    const urlWithId = `${url}/${id}`;
    return axiosClient.delete(urlWithId);
  },
};

export default todoApi;
