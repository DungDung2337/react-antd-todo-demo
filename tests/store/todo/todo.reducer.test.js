import {
  removeTodoSuccess,
  toggleTodoStatusSuccess,
} from '../../../src/store/todo/actions';
import {
  initialState,
  todoReducer,
} from '../../../src/store/todo/reducers/todo.reducer';

import { createTestTodo } from './factories/todo.factory';

describe('todo reducer', () => {
  it('should handle addTodo', () => {
    expect(
      todoReducer(
        initialState,
        createTestTodo({
          id: '440351',
          name: 'This is a test todo',
          completed: true,
        })
      )
    ).toMatchSnapshot();
  });

  it('should handle removeTodo', () => {
    const createdTodoAction = createTestTodo({});

    let todoReducerState = todoReducer(initialState, createdTodoAction);

    expect(todoReducerState).toBeDefined();
    expect(todoReducerState.todos.data).toHaveLength(1);

    todoReducerState = todoReducer(
      initialState,
      removeTodoSuccess(createdTodoAction.payload)
    );

    expect(todoReducerState.todos.data).toHaveLength(0);
  });

  it('should handle toggleTodoStatus', () => {
    const createTodoAction = createTestTodo({
      id: '440351',
    });

    const state = todoReducer(initialState, createTodoAction);

    const todoReducerState = todoReducer(
      state,
      toggleTodoStatusSuccess(createTodoAction.payload)
    );

    expect(todoReducerState.todos.data[0].completed).toBe(
      !createTodoAction.payload.completed
    );
  });
});
