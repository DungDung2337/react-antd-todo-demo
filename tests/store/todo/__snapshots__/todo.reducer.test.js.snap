// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo reducer should handle addTodo 1`] = `
Object {
  "errors": null,
  "loading": false,
  "loadingForm": false,
  "todos": Object {
    "data": Array [
      Object {
        "completed": true,
        "id": "440351",
        "name": "This is a test todo",
      },
    ],
    "pagination": Object {
      "_limit": 3,
      "_page": 1,
      "_totalRows": 1,
    },
  },
}
`;
